src/modules/position-creator/services/position.creator.transaction.ts

```diff
@@ -0,0 +1,467 @@
+        private readonly stakingAbi: StakingAbiService,
+        private readonly stakingProxyAbi: StakingProxyAbiService,
+        private readonly tokenService: TokenService,
+        private readonly mxProxy: MXProxyService,
+    ) {}
+
+    async createLiquidityPositionSingleToken(
+        sender: string,
+        pairAddress: string,
+        payment: EsdtTokenPayment,
+        tolerance: number,
+    ): Promise<TransactionModel> {
+        const uniqueTokensIDs = await this.tokenService.getUniqueTokenIDs(
+            false,
+        );
+
+        if (!uniqueTokensIDs.includes(payment.tokenIdentifier)) {
+            throw new Error('Invalid ESDT token payment');
+        }
+
+        const singleTokenPairInput =
+            await this.posCreatorCompute.computeSingleTokenPairInput(
+                pairAddress,
+                payment,
+                tolerance,
+            );
+
+        const contract = await this.mxProxy.getPostitionCreatorContract();
+
+        return contract.methodsExplicit
+            .createLpPosFromSingleToken([
+                new AddressValue(Address.fromBech32(pairAddress)),
+                new BigUIntValue(singleTokenPairInput.amount0Min),
+                new BigUIntValue(singleTokenPairInput.amount1Min),
+                ...singleTokenPairInput.swapRouteArgs,
+            ])
+            .withSingleESDTTransfer(
+                TokenTransfer.fungibleFromBigInteger(
+                    payment.tokenIdentifier,
+                    new BigNumber(payment.amount),
+                ),
+            )
+            .withSender(Address.fromBech32(sender))
+            .withGasLimit(gasConfig.positionCreator.singleToken)
+            .withChainID(mxConfig.chainID)
+            .buildTransaction()
+            .toPlainObject();
+    }
+
+    async createFarmPositionSingleToken(
+        sender: string,
+        farmAddress: string,
+        payments: EsdtTokenPayment[],
+        tolerance: number,
+    ): Promise<TransactionModel> {
+        const [pairAddress, farmTokenID, uniqueTokensIDs] = await Promise.all([
+            this.farmAbiV2.pairContractAddress(farmAddress),
+            this.farmAbiV2.farmTokenID(farmAddress),
+            this.tokenService.getUniqueTokenIDs(false),
+        ]);
+
+        if (!uniqueTokensIDs.includes(payments[0].tokenIdentifier)) {
+            throw new Error('Invalid ESDT token payment');
+        }
+
+        for (const payment of payments.slice(1)) {
+            if (payment.tokenIdentifier !== farmTokenID) {
+                throw new Error('Invalid farm token payment');
+            }
+        }
+
+        const singleTokenPairInput =
+            await this.posCreatorCompute.computeSingleTokenPairInput(
+                pairAddress,
+                payments[0],
+                tolerance,
+            );
+
+        const contract = await this.mxProxy.getPostitionCreatorContract();
+
+        return contract.methodsExplicit
```

Comment:
use methods instead as you dont have to explicitly construc the types

Reply:
will stick with explicit methods for now giving a better control of how encoding is done